packages/core/src/utils.ts

```diff
@@ -92,6 +93,16 @@ export async function copyWalletAddress(text: string): Promise<void> {
   }
 }
 
+export const toHexString = (val: number | string): string => typeof val === 'number' ? `0x${val.toString(16)}` : val
+
+export function chainIdToHex(chains : Chain[] | ChainWithDecimalId[] ): 
```

Comment:
if change the types to `type ChainId = number | string` then we could have just `Chain[]` vs `Chain[] | ChainWithDecimalId[]`